packages/solid/store/src/store.ts

```diff
@@ -39,21 +39,33 @@ function wrap<T extends StoreNode>(value: T, name?: string): DeepReadonly<T> {
   return p;
 }
 
-export function isWrappable(obj: any) {
-  return (
+function isUnwrappable(obj: {} | undefined | null): obj is NotWrappable {
+  return !(
     obj != null &&
     typeof obj === "object" &&
-    (obj[$PROXY] || !obj.__proto__ || obj.__proto__ === Object.prototype || Array.isArray(obj))
+    (
+      obj[$PROXY as keyof typeof obj] ||
+      !(obj as any).__proto__ ||
+      (obj as any).__proto__ === Object.prototype ||
+      Array.isArray(obj)
+    )
   );
 }
 
-export function unwrap<T extends StoreNode>(item: any, set = new Set()): T {
+export function isWrappable<T>(obj: T) {
+  return !isUnwrappable(obj);
```

Comment:
Also, there isn't a `is not Type` so we use this (NonNullable introduces the `never` and we don't want that).

Reply:
You can do `export function isWrappable<T>(obj: T | NotWrappable): obj is T` for `is not T`.